File: server/data/translation_bench/prime_sieve.py
def sieve(limit):
    flags = []
    for i in range(limit + 1):
        flags.append(True)
    flags[0] = False
    if limit >= 1:
        flags[1] = False
    i = 2
    while i * i <= limit:
        if flags[i]:
            j = i * i
            while j <= limit:
                flags[j] = False
                j = j + i
        i = i + 1
    primes = []
    for k in range(limit + 1):
        if flags[k]:
            primes.append(k)
    return primes


def main():
    result = sieve(50)
    line = ""
    for value in result:
        line = line + str(value) + " "
    print(line.strip())


main()

File: server/data/results/translations/lite/plain/go/prime_sieve.go
package main

import (
	"fmt"
	"strings"
)

func sieve(limit int) []int {
	flags := make([]bool, limit+1)
	for i := 0; i <= limit; i++ {
		flags[i] = true
	}
	if limit >= 0 {
		flags[0] = false
	}
	if limit >= 1 {
		flags[1] = false
	}
	i := 2
	for i*i <= limit {
		if flags[i] {
			j := i * i
			for j <= limit {
				flags[j] = false
				j = j + i
			}
		}
		i = i + 1
	}
	var primes []int
	for k := 0; k <= limit; k++ {
		if flags[k] {
			primes = append(primes, k)
		}
	}
	return primes
}

func main() {
	result := sieve(50)
	line := ""
	for _, value := range result {
		line = line + fmt.Sprintf("%d", value) + " "
	}
	fmt.Println(strings.TrimSpace(line))
}